src/utils/array_utils.F90.in

```diff
@@ -39,15 +40,13 @@ contains
     ! Returns the indices of the maximum k values in data, in sorted order (so the first
     ! index will be the largest, the second index will be the second largest, etc.)
     !
-    ! The output array (max_indices) is allocated here
-    !
-    ! This is O(n*k) (where n is the size of the input array)
+    ! The output array (max_indices) should be of size k
     !
     ! !ARGUMENTS:
-    integer  , intent(in)               :: lb ! lower bound of data array
-    real(r8) , intent(in)               :: data(lb:)
-    integer  , intent(in)               :: k  ! number of max indices to find
-    integer  , allocatable, intent(out) :: max_indices(:)
+    integer  , intent(in)  :: lb ! lower bound of data array
+    real(r8) , intent(in)  :: data(lb:)
+    integer  , intent(in)  :: k  ! number of max indices to find
+    integer  , intent(inout) :: max_indices(:)
```

Comment:
This should still be intent(out). It doesn't matter hugely, but it better documents the fact that the whole array is written, and the initial values are irrelevant.

Reply:
Done.

100% of unit tests PASS.